spectral_cube/tests/test_subcubes.py

```diff
@@ -150,3 +150,17 @@ def test_ds9region_new(regfile, result):
     #region = 'circle(2,2,2)'
     #subcube = cube.subcube_from_ds9region(region)
 
+
+@pytest.mark.skipif('not regionsOK', reason='Could not import regions')
+def test_regions():
+    cube, data = cube_and_raw('adv.fits')
+
+    # content of image.reg
+    regpix = regions.RectanglePixelRegion(regions.PixCoord(0.5, 1), width=4, height=2)
```

Comment:
This rectangle goes outside the image, which is OK but not great.  `adv.fits` is pretty small, so you want a small region.  This region will go from xlo,xhi,ylo,yhi = [-1.5, 2.5, 0, 2] = [0.5-4/2, 0.5+4/2, 1-2/2, 2+2/2].  So try to pick a pixel coordinate and width/height that fit in the cube better.

Reply:
BUT, it would also be good to have at least three tests: one where the region is in the cube, one where it partly overlaps with the cube, and one where it's outside the cube and therefore fails.

Reply:
this test is just for checking the extraction of the spectral part. the tests above checks for the spatial one.But, if you want I will do it.

Reply:
OK, that's fine, but the region selection is happening both spatially and spectrally.  As long as you've done the by-hand math to check that the indices below match the selections from the region, it's OK.    You should label this test `_spectral` if it's intended to test only spectral

Reply:
ya, I better put that `_spectral` label!